### DISCUSSION TITLE
Port forwarding in Lima for network

### DISCUSSION THREAD

**Discussion by wackazong** [abiosoft/colima#934] [General]
I set up a Kind Kubernetes Cluster in Colima according to https://opencredo.com/blogs/building-the-best-kubernetes-test-cluster-on-macos/

I want to use port forwarding to forward the local host ports 80 and 443 to a specific IP in the Kind network. How can I do that?

Goal is to be able to use the `.localhost` domain in Chrome for development.

**Comment by abiosoft** [comment]
The port forwarding is automatic and there is no special workaround required to listen on ports 80 and 443 with Colima.
As long as you have a Kubernetes service of type `NodePort` or `LoadBalancer` listening on those ports, it should just work.

**Comment by wackazong** [comment]
Thing is, I am not running k3s but kind in order to be able to use a multi node config, as described in the link I posted. With that, port forwarding does not work automatically.

So either I need to move to the built-in k3s (AFAIK I cannot use multiple nodes there) or I need to manually set up port forwarding for the kind network.

How to do that was my question. Any pointers very much appreciated. Thanks!

**Reply by wackazong** [reply]
I tried adding this to ~/.lima/_config/override.yaml but it does not have the desired effect:

```yaml
portForwards:
  - guestIP: 172.18.255.200
    guestPort: 80
    hostIP: 127.0.0.1
    hostPort: 80
    proto: tcp
```

What I want to accomplish: forward my local port 80 of the host to 172.18.255.200:80 in the VM so that I can reach the kubernetes load balancer IP from my host via localhost:80

**Comment by wackazong** [comment]
I managed to set up port forwarding manually using an ssh tunnel. But this is kind of a hacky workaround IMHO.

**Comment by abiosoft** [comment]
Let me get your exact scenario. 

You are utilising Kind for a multi-node Kuberntes cluster and would like to selectively forward some ports from the cluster to the host. Right?

**Comment by wackazong** [comment]
Yes. What I do with ssh is:

```console
sudo ssh -F ~/.colima/ssh_config colima -L 80:172.18.255.200:80 -L 443:172.18.255.200:443
```

Where `172.18.255.200` is the IP of the Loadbalancer Service in the Kind Kubernetes cluster. Then I can access my local dev cluster from the host at localhost port 80 and 443. This means I can use urls like my-cluster.localhost in chrome.

**Comment by wackazong** [comment]
I solved it with `socat` for now: https://github.com/lima-vm/lima/discussions/2087
